hazelcast/include/hazelcast/client/serialization/pimpl/compact.h

```diff
@@ -411,6 +411,96 @@ class HAZELCAST_API compact_reader
     boost::optional<double> read_nullable_float64(
       const std::string& field_name);
 
+    /**
+     * Reads a nullable array of nullable booleans.
+     *
+     * @param field_name name of the field.
+     * @return the value of the field.
+     * @throws hazelcast_serialization if the field does not exist in the
+     * schema or the type of the field does not match with the one defined
+     * in the schema.
+     */
+    boost::optional<std::vector<boost::optional<bool>>>
+    read_array_of_nullable_boolean(const std::string& field_name);
+
+    /**
+     * Reads a nullable array of nullable 8-bit two's complement signed
+     * integers.
+     *
+     * @param field_name name of the field.
+     * @return the value of the field.
+     * @throws hazelcast_serialization if the field does not exist in the
+     * schema or the type of the field does not match with the one defined
+     * in the schema.
+     */
+    boost::optional<std::vector<boost::optional<int8_t>>>
+    read_array_of_nullable_int8(const std::string& field_name);
+
+    /**
+     * Reads a nullable array of nullable 16-bit two's complement signed
+     * integers.
+     *
+     * @param field_name name of the field.
+     *  @return the value of the field.
+     *  @throws hazelcast_serialization if the field does not exist in the
+     *  schema or the type of the field does not match with the one defined
+     *  in the schema.
```

Comment:
```suggestion
     * @return the value of the field.
     * @throws hazelcast_serialization if the field does not exist in the
     * schema or the type of the field does not match with the one defined
     * in the schema.
```

Similar for other methods